src/routers/SwitchRouter.js

```diff
@@ -258,10 +295,12 @@ export default (routeConfigs, config = {}) => {
       }
 
       if (activeChildIndex !== state.index) {
-        return getNextState(prevState, {
-          ...state,
-          index: activeChildIndex,
-        });
+        return updateRouteKeyHistory(
```

Comment:
same as above re: moving to getNextState

Reply:
done